server/paymentProviders/opencollective/host.js

```diff
@@ -12,6 +15,26 @@ paymentMethodProvider.features = {
   waitToCharge: false,
 };
 
+paymentMethodProvider.refundTransaction = async (transaction, user) => {
+  if (transaction.CollectiveId === transaction.FromCollectiveId) {
+    throw new Error('Cannot refund a transaction from the same collective');
+  }
+
+  const host = await models.Collective.findByPk(transaction.CollectiveId);
```

Comment:
Why would the `host` be matching `transaction.CollectiveId`. That doesn't make sense.

Reply:
```suggestion
  const collective = await models.Collective.findByPk(transaction.CollectiveId);
```

Reply:
There is no guarantee that we are looking at a CREDIT and the Collective is the one where the money was added, look at how the `collective` paymentMethod is handling this.

Reply:
Or look at createRefundTransaction,

```
  if (transaction.type === DEBIT) {
    transaction = await transaction.getRelatedTransaction({ type: CREDIT });
  }
```

Reply:
Thanks, I've added this condition and checked to see if a `CREDIT` transaction exists. Let me know if I've missed anything> 🤔